zum/requests/core.py

```diff
@@ -19,5 +51,15 @@ def generate_request(raw_endpoint: Dict[str, Any], arguments: List[str]) -> Requ
     headers, remaining_arguments = reduce_arguments(
         raw_endpoint.get("headers"), remaining_arguments
     )
-    body, _ = reduce_arguments(raw_endpoint.get("body"), remaining_arguments)
+
+    body: Dict[str, Any] = dict()
+    if (
+        raw_endpoint.get("body") is not None
+        and isinstance(raw_endpoint.get("body"), str)
+        and raw_endpoint.get("body") == "json"
+    ):
+        body = load_request_body(remaining_arguments)
+    else:
+        body, _ = reduce_arguments(raw_endpoint.get("body"), remaining_arguments)
```

Comment:
I think that this whole piece of code could be decoupled and implemented as a method that receives the output of calling `raw_endpoint.get("body")` and the `remaining_arguments` and returns the corresponding body. That way, this new function could operate all the conditionals and all that stuff then execute the `load_request_body` method directly with the first argument (if it exists)